EverestAPI/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using EverestAPI.Models;
+using EverestAPI.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EverestAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomerController : ControllerBase
+    {
+        private readonly ICustomerService _customerService;
+
+        public CustomerController(ICustomerService customerService)
+        {
+            _customerService = customerService ?? throw new ArgumentNullException(nameof(customerService));
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            var response = _customerService.GetAll();
+
+            return Ok(response);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetById(long id)
+        {
+            try
+            {
+                var response = _customerService.GetById(id);
+
+                return Ok(response);
+            }
+            catch (ArgumentNullException exception)
+            {
+                var message = exception.InnerException?.Message ?? exception.Message;
+
+                return NotFound(message);
+            }
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] CustomerModel customerModel)
+        {
+            try
+            {
+                _customerService.Create(customerModel);
+
+                return Created("Id:", customerModel.Id);
+            }
+            catch(ArgumentException exception)
+            {
+                var message = exception.InnerException?.Message ?? exception.Message;
+
+                return BadRequest(message);
+            }
+        }
+
+        [HttpPut("{id}")]
+        public IActionResult Update(CustomerModel customerModel)
```

Comment:
```suggestion
        public IActionResult Update(CustomerModel customerModel, long id)
```
Você poderia capturar o id que vem da rota e utilizar ele para realizar a pesquisa do Customer a ser atualizado. Acredito que seja melhor que passar o id na rota e passar o id no body (corpo da requisição).